src/inference/dev_api/memory_solver.hpp

```diff
@@ -147,7 +147,7 @@ class MemorySolver {
                     for (auto* box_in_slot : time_slots[i_slot]) {
                         // intersect with already stored boxes for all covered time slots
                         // and move up the new one if needed
-                        popped_up = popped_up || popupTogetherWith(box, *box_in_slot);
+                        popped_up = popped_up | popupTogetherWith(box, *box_in_slot);
```

Comment:
can we just change to
```suggestion
                        popped_up = popupTogetherWith(box, *box_in_slot) | popped_up;
```
and add some comment that we need 'popupTogetherWith' to be called even if `popped_up` is already 'true'. Also it will not produce warning that 'bitwise OR' is called for two booleans ( 2 & 1 = 0, but 2 && 1 = true)